torch/ao/quantization/fx/prepare.py

```diff
@@ -162,6 +162,13 @@ def is_output_dtype_supported_by_backend(
     return output_dtype is None or \
         output_dtype == node_name_to_target_dtype[node.name]["output_activation_dtype"]
 
+def need_remove_observer(node, modules, node_name_to_target_dtype):
```

Comment:
can we add some comments on what this is doing and why?

Reply:
would it be possible to name this function after what it is checking?  `need_remove_observer` is pretty generic and it may not be clear on why exactly this function is there.

I think a PR summary explaining the issue and the fix might help me give better comments here, right now I'm not yet sure what exactly is broken.